### ISSUE TITLE
19 Nov: Subsetting data in R

### ISSUE THREAD

**Issue by murraycadzow** [smilefreak/studyGroup#45]
This week learn how to subset data in R using subset(), [ ] and dplyr

I'll cover some boolean logic and a very brief intro to dplyr, with a focus on how to subset

NB: if you don't have dplyr installed in R run: install.packages("dplyr")

**Comment by TomKellyGenetics** [comment]
Great session Murray, great content and delivery. Especially pace and solid coverage of the fundamentals of booleans.

Just a few pointers:

I'd be careful assuming knowledge of prior workshops (e.g., ggplot) in case it makes newcomers feel isolated.

If we're targeting new users, I wouldn't worry about efficiency or performance too much. Experienced users will know R isn't the best tool if that's what they're after anyway. Although I understand that aspect of yesterdays session was answering a discipline-specific example / question.

**Comment by murraycadzow** [comment]
Thanks for the feedback Tom.

I think that while not ideal to assume knowledge of prior workshops, it can be done so long as it is more of an extra side topic to show how past lessons tie into the current content. Or given that some lessons will by there nature require an understanding of basics we should endeavour to outline this in the event information, e.g. the ggplot2 lesson assumes an understanding of how to use R to start with.

However, I should have been a bit more explicit for people that might not have been to previous lessons in pointing out all that previous lessons content could be found in the lessons section of the repo

With regard to efficiency, while my example was quite far from an everyday occurrence  it is useful to make everyone aware that there are good ways to do things and bad ways and to start thinking about writing good code from the start eg only putting what actually needs to be done inside of a loop

**Comment by TomKellyGenetics** [comment]
Yep, was a good idea to reinforce to ggplot session for our regulars (and tie in the "Hadely-verse"). It's a hard balance to strike. Not a big deal, was just a bit worried we might have lost some of the new people I poked to come near the end. Cool figures always help with concepts / motivation / relevance though :)

More complex examples are ok, just make sure it's clear they're an extension topic (perhaps a little less emphasis on it, that's all). Starting with good habits is a great idea and why I encouraged newbies to come to this session, I wish I'd learnt `dplyr` to start with, nested commands are a pain to read and a tricky habit to kick.

Really good session really, these are very minor points to improve. Great topic choice, this approach could really belong in basic R sessions rather than presented as an advanced add-on. There's a lot of debate around this in the software carpentry material, mainly because the "gapminder" lessons draw a lot from the Hadley-verse and use R-specific techniques in contrast to the existing core-R lessons which match the core-python content. Something we also have luxury of testing out in the study group format.
